File: 114.cpp
#include<iostream>

long long F(int m, int n)
{

    long long sols = 1;

    if(m>n)
    {
        return sols;
    }

    for(int spos = 0; spos <= n-m; spos++)
    {
        for(int blen = m; blen <= n - spos; blen++)
        {
            sols += F(m, n - spos - blen - 1);
        }

    }

    return sols;
}

int main(){

    std::cout << F(3,50) << std::endl;

}

File: 114.py
#! /usr/bin/python end

def F(m, n):
    sol = 1

    if(n>m):
        return sol
    
    for spos in xrange(m-n+1):
        for blen in xrange(n,m-spos+1):
            sol += F(m - spos - blen - 1, n)
    
    return sol

print F(40,3)
